lib/ansible/template/__init__.py

```diff
@@ -1004,58 +1005,58 @@ def _query_lookup(self, name, *args, **kwargs):
     def _lookup(self, name, *args, **kwargs):
         instance = self._lookup_loader.get(name, loader=self._loader, templar=self)
 
-        if instance is not None:
-            wantlist = kwargs.pop('wantlist', False)
-            allow_unsafe = kwargs.pop('allow_unsafe', C.DEFAULT_ALLOW_UNSAFE_LOOKUPS)
-            errors = kwargs.pop('errors', 'strict')
+        if instance is None:
+            raise AnsibleError("lookup plugin (%s) not found" % name)
 
-            from ansible.utils.listify import listify_lookup_plugin_terms
-            loop_terms = listify_lookup_plugin_terms(terms=args, templar=self, loader=self._loader, fail_on_undefined=True, convert_bare=False)
-            # safely catch run failures per #5059
-            try:
-                ran = instance.run(loop_terms, variables=self._available_variables, **kwargs)
-            except (AnsibleUndefinedVariable, UndefinedError) as e:
-                raise AnsibleUndefinedVariable(e)
-            except Exception as e:
-                if self._fail_on_lookup_errors:
-                    msg = u"An unhandled exception occurred while running the lookup plugin '%s'. Error was a %s, original message: %s" % \
-                          (name, type(e), to_text(e))
-                    if errors == 'warn':
-                        display.warning(msg)
-                    elif errors == 'ignore':
-                        display.display(msg, log_only=True)
-                    else:
-                        raise AnsibleError(to_native(msg))
-                ran = [] if wantlist else None
+        wantlist = kwargs.pop('wantlist', False)
+        allow_unsafe = kwargs.pop('allow_unsafe', C.DEFAULT_ALLOW_UNSAFE_LOOKUPS)
+        errors = kwargs.pop('errors', 'strict')
 
-            if ran and not allow_unsafe:
-                if wantlist:
-                    ran = wrap_var(ran)
+        loop_terms = listify_lookup_plugin_terms(terms=args, templar=self, loader=self._loader, fail_on_undefined=True, convert_bare=False)
+        # safely catch run failures per #5059
+        try:
+            ran = instance.run(loop_terms, variables=self._available_variables, **kwargs)
+        except (AnsibleUndefinedVariable, UndefinedError) as e:
+            raise AnsibleUndefinedVariable(e)
+        except Exception as e:
+            if self._fail_on_lookup_errors:
```

Comment:
`self._fail_on_lookup_errors` appears to be always `True`. Can we remove it or do we keep it for future and/or 3rd party Templars?